### PR TITLE
Delegate to ImageDecoder if >= P

### PR DESCRIPTION
**Pull request by jrodbx**

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso3/RequestHandler.java [hunk 1]
```diff
@@ -209,18 +221,49 @@ static void calculateInSampleSize(int reqWidth, int reqHeight, int width, int he
...
+
+    return decodeStreamPreP(request, bufferedSource);
+  }
+
+  @TargetApi(28)
+  @SuppressLint("Override")
+  private static Bitmap decodeStreamP(final Request request, BufferedSource bufferedSource)
+      throws IOException {
+    ImageDecoder.Source imageSource =
+        ImageDecoder.createSource(ByteBuffer.wrap(bufferedSource.readByteArray()));
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: Ugh

File: picasso/src/main/java/com/squareup/picasso3/RequestHandler.java [hunk 2]
```diff
@@ -209,18 +221,49 @@ static void calculateInSampleSize(int reqWidth, int reqHeight, int width, int he
...
     BufferedSource bufferedSource = Okio.buffer(source);
 
+    if (Build.VERSION.SDK_INT >= 28) {
+      return decodeStreamP(request, bufferedSource);
+    }
+
+    return decodeStreamPreP(request, bufferedSource);
+  }
+
+  @TargetApi(28)
```
[the remark this answers is not in the document]
Comment (jrodbx) [reply picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: I went with that first, but hit and filed this Lint issue: https://issuetracker.google.com/issues/77246433

[the remark this answers is not in the document]
Comment (jrodbx) [reply picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: oh right, but then I added the SuppressLint to shut that up.  nvm.  will update.

File: picasso/src/main/java/com/squareup/picasso3/RequestHandler.java (hunk 1 above)
Comment (jrodbx) [reply picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: I know...

Comment (jrodbx) [reply picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: Looks like there's a perfectly valid, yet hidden overload here: https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/graphics/java/android/graphics/ImageDecoder.java#900

🤔

Comment (NightlyNexus) [reply picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: why does ImageDecoder not have a streaming API?!

Comment (JakeWharton) [reply picasso/src/main/java/com/squareup/picasso3/RequestHandler.java]: It needs to read multiple times and have random access

**Pull request merged**